Shared/Samples/Filter building scene layer/FilterBuildingSceneLayerView.swift

```diff
@@ -0,0 +1,208 @@
+                        guard let fullModelSublayer = buildingSceneLayer.sublayers
+                            .first(where: { $0.name == "Full Model" }) as? BuildingGroupSublayer else { return }
+                        
+                        groupSublayers = fullModelSublayer.sublayers
+                            .compactMap { $0 as? BuildingGroupSublayer }
+                        
+                        // Gets the attribute statistics to get the floor
+                        // information.
+                        let statistics = try await buildingSceneLayer.statistics
+                        
+                        // Gets the floor statistics.
+                        guard let floorStatistics = statistics[.floorFieldKey] else { return }
+                        
+                        // Gets all the available floors and sorts the floors
+                        // from top to bottom floor.
+                        availableFloors = floorStatistics.mostFrequentValues.sorted { $0 > $1 } + [.allFloorsLabel]
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .task(id: tapPoint) {
+                    // Clears any previous selection if there was a new
+                    // tap on the screen.
+                    selectedSublayer?.clearSelection()
+                    
+                    guard let tapPoint, let buildingSceneLayer else { return }
+                    
+                    let result = try? await proxy
+                        .identify(on: buildingSceneLayer, screenPoint: tapPoint, tolerance: 5)
+                    
+                    let sublayerResult = result?.sublayerResults
+                        // Get the first sublayer result that has geo elements.
+                        .first(where: { !$0.geoElements.isEmpty })
+                    
+                    guard let sublayerResult,
+                          let feature = sublayerResult.geoElements.first as? Feature,
+                          let component = sublayerResult.layerContent as? BuildingComponentSublayer
+                    else { return }
+                    
+                    component.selectFeature(feature)
+                    popup = sublayerResult.popups.first
+                    selectedSublayer = component
+                }
+        }
+        .errorAlert(presentingError: $error)
+        .toolbar {
+            ToolbarItem(placement: .bottomBar) {
+                Button("Settings", systemImage: "gear") {
+                    settingsAreVisible = true
+                }
+                .popover(isPresented: $settingsAreVisible) { [settings] in
+                    settings
+                        .frame(idealWidth: 400, idealHeight: 500)
+                        .presentationCompactAdaptation(.popover)
+                }
+            }
+        }
+        .popover(isPresented: $showPopup, attachmentAnchor: .point(.top)) { [popup] in
```

Comment:
It appears that `popup` does not need to be captured here.

Reply:
That's even more inexplicable!

Reply:
@nixta has a similar issue with a FeatureForm demo he was creating. He is using a `.sheet` to display the feature form, but the first time the sheet was displayed, the `form` was nil. The second time `form` was not nil and worked fine. Capturing `form`, as below, caused it to work every time:

```
        .sheet(isPresented: $showingForm) { [form] in  //<- capturing form here fixes Nick's issue
            if let form {
                FeatureFormView(root: form, isPresented: $showingForm)
            } else {
                Text("What! No Form!!")
            }
        }
```